hydra/grammar/OverrideLexer.g4

```diff
@@ -4,28 +4,50 @@
 // If you make changes here be sure to update the documentation (and update the grammar in command_line_syntax.md)
 lexer grammar OverrideLexer;
 
-EQUAL: '=';
+
+// Re-usable fragments.
+fragment DIGIT: [0-9_];
+fragment WS_: [ \t]+;
+
+/////////
+// KEY //
+/////////
+
+mode KEY;
+
+EQUAL: '=' WS_? -> mode(ARGS);
+
 TILDE: '~';
 PLUS: '+';
 AT: '@';
 COLON: ':';
 ATCOLON: '@:';
 SLASH: '/';
-PARENTHESIS_OPEN: '(';
-COMMA: ',';
-PARENTHESIS_CLOSE: ')';
-DASH: '-';
-BACKSLASH: '\\';
-DOT: '.';
-DOLLAR: '$';
-STAR: '*';
-BRACKET_OPEN: '[';
-BRACKET_CLOSE: ']';
-BRACE_OPEN: '{';
-BRACE_CLOSE: '}';
+
+fragment CHAR: [a-zA-Z];
+ID: (CHAR|'_') (CHAR|DIGIT|'_')*;
+fragment LIST_INDEX: '0' | [1-9][0-9]*;
+DOT_PATH: (ID | LIST_INDEX) ('.' (ID | LIST_INDEX))+;
+
+WS: WS_ -> skip;
```

Comment:
what is the use case of skipping whitespace in KEY mode?

Reply:
Backward compatibility with current grammar in master. But it could definitely be removed if you prefer.

Reply:
If there are no tests that ensuring it we can safely ignore it. I can't think of a use case for whitespace in key right now.

Reply:
> If there are no tests that ensuring it we can safely ignore it. I can't think of a use case for whitespace in key right now.

There were no tests but I just added some earlier ;) I changed it to forbid whitespaces in KEY mode (and changed these tests accordingly) => this is in 977bca69